vllm/model_executor/layers/mamba/ops/causal_conv1d.py

```diff
@@ -11,76 +11,78 @@ def causal_conv1d_fn(
     activation: either None or "silu" or "swish"
 
     out: (batch, dim, seqlen)
     """
     if activation not in [None, "silu", "swish"]:
         raise NotImplementedError("activation must be None, silu, or swish")
-    if x.stride(2) != 1 and x.stride(1) != 1:
+    if x.stride(-1) != 1:
         x = x.contiguous()
     bias = bias.contiguous() if bias is not None else None
-    if seq_idx is not None:
-        assert (initial_states is
-                None), "initial_states must be None if seq_idx is not None"
-        assert (not return_final_states
-                ), "If seq_idx is not None, we don't return final_states_out"
-    seq_idx = seq_idx.contiguous() if seq_idx is not None else None
-    if initial_states is not None and (initial_states.stride(2) != 1
-                                       and initial_states.stride(1) != 1):
-        initial_states = initial_states.contiguous()
-    if return_final_states:
-        assert (
-            x.stride(1) == 1
-        ), "Only channel-last layout support returning final_states_out"
-        if final_states_out is not None:
-            assert (final_states_out.stride(2) == 1
-                    or final_states_out.stride(1) == 1)
-        else:
-            batch, dim, seqlen = x.shape
-            width = weight.shape[1]
-            final_states_out = torch.empty(batch,
-                                           width - 1,
-                                           dim,
-                                           device=x.device,
-                                           dtype=x.dtype).transpose(1, 2)
-    else:
-        final_states_out = None
 
-    out = ops.causal_conv1d_fwd(x, weight, bias, seq_idx, initial_states,
-                                final_states_out, activation
+    if conv_states is None:
+        conv_states = torch.empty(x.shape[0],
+                                  x.shape[1],
+                                  weight.shape[1] - 1,
+                                  device=x.device,
+                                  dtype=x.dtype)
+
+    out = ops.causal_conv1d_fwd(x, weight, bias, conv_states, cu_seq_len,
+                                cache_indices, has_initial_state, activation
                                 in ["silu", "swish"])
-    return (out, None) if not return_final_states else (out, final_states_out)
+    return (out, conv_states)
 
 
-def causal_conv1d_update(x: torch.Tensor,
-                         conv_state: torch.Tensor,
-                         weight: torch.Tensor,
-                         bias: Optional[torch.Tensor] = None,
-                         activation: Optional[str] = None):
+def causal_conv1d_update(x,
+                         conv_state,
+                         weight,
+                         bias=None,
+                         activation=None,
+                         cache_seqlens=None):
     """
-    x: (batch, dim)
-    conv_state: (batch, dim, width)
+    x: (batch, dim) or (batch, dim, seqlen)
```

Comment:
What happens when the seq_lens aren't all identical?

Reply:
I'd taken this code from  https://github.com/Dao-AILab/causal-conv1d/commit/2e099719ccdb7ac30d95211971f8db2a44507c47 , I'm not sure if this scheme here is supposed to support varlen or not.
I thought it would be a good idea to update the `causal_conv1d_update` kernel with the up to date causal_conv1d code . 
I also don't see any implementation that uses this functionality in the `mamba_ssm` repo